grails-app/controllers/org/pih/warehouse/api/LocationApiController.groovy

```diff
@@ -146,6 +147,11 @@ class LocationApiController extends BaseDomainApiController {
             location.supportedActivities.clear()
         }
 
+        // If the organization chosen for the created location is inactive, throw an exception
+        if (!location.organization.active) {
```

Comment:
Organization is nullable so you'll want to use null protection (?.) here. 

Can you explain why we need this validation here? Won't we get this for free from the validation in the security filters? I assume it's because you need to throw an exception here because otherwise the API request is going through the security filter, getting the session.warehouse attribute set to null and then going on its way. Whereas we need the API request to fail. However, I believe we need requests that are associated with Location that is disabled should always fail. Whether they are creating, editing, deleting the location.

And I assume we don't necessarily want every API endpoint to have to add this check. A more generic way of handling this would be to have the security filter check whether the request is a web request or API request and then handle the response appropriately. This way we keep all of the authorization / validation in the same place. 
```
// ignore the name of the method, this was primarily used for ajax request but i think it works for API requests as well.
if (RequestUtil.isAjax(request)) {
    throw new IllegalArgumentException("The organization ${location.organization.name} is inactive, you can't assign it to the location")
} else { 
    flash.message = "Your location is disabled (either the associated organization or your location itself became inactive)"
}
```

Reply:
The validation in the security filters only checks if your current location's organization is inactive - the validation here checks if you are creating a location and trying to assign an inactive organization to it (you are not logged in to the location you are creating, so the validation in `SecurityFilters` has nothing to this one in my opinion)